docs/standard/native-interop/qualify-net-types-for-interoperation.md

```diff
@@ -40,10 +42,237 @@ If you intend to expose types in an assembly to COM applications, consider the r
+
+Many more customization options exist and they are mentioned in the [Applying Interop Attributes](apply-interop-attributes.md) documentation, but these guidelines provide the minimum requirements for common scenarios.
+
+### Defining COM interfaces in .NET
+
+When .NET code tries to call a method on a COM object through an interface with the <xref:System.Runtime.InteropServices.ComImportAttribute>, it needs to build up a virtual function table (also known as vtable or vftable) to from the .NET definition of the interface to determine the native code to call. This process is very complex, but we can look at some simple cases to see how it works.
+
+We will start with a COM interface with a few methods:
+
+```c++
+struct IComInterface : public IUnknown
+{
+    STDMETHOD(Method)() = 0;
+    STDMETHOD(Method2)() = 0;
+};
+```
+
+For this interface, the following table describes its virtual function table layout:
+
+| `IComInterface` Virtual Function Table slot | Method name |
+|-----------------------------|-------------|
+| 0 | `IUnknown::QueryInterface` |
+| 1 | `IUnknown::AddRef` |
+| 2 | `IUnknown::Release` |
+| 3 | `IComInterface::Method` |
+| 4 | `IComInterface::Method2` |
+
+As we see, each method is added to the virtual function table in the order it was declared. The particular order is defined by the C++ compiler, but for simple cases without overloads, declaration order defines the order in the table.
+
+To declare a .NET interface that corresponds to this interface, we would declare it as follows:
+
+```csharp
+[ComImport]
+[InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+[Guid(/* The IID for IComInterface */)]
+interface IComInterface
+{
+    void Method();
+    void Method2();
+}
+```
+
+The <xref:System.Runtime.InteropServices.InterfaceTypeAttribute> specifies the base interface. It provides a few options:
+
+| <xref:System.Runtime.InteropServices.ComInterfaceType> Value | Base Interface Type | Behavior for members on attributed interface |
+|------------|---------------|------------------|
+| `InterfaceIsIUnknown` | `IUnknown` | Virtual function table first has the members of `IUnknown`, then the members of this interface in declaration order. |
+| `InterfaceIsIDispatch` | `IDispatch` | Members are not added to the virtual function table. They are only accessible through `IDispatch` |
+| `InterfaceIsDual` | `IDispatch` | Virtual function table first has the members of `IDispatch`, then the members of this interface in declaration order. |
+| `InterfaceIsIInspectable` | `IInspectable` | Virtual function table first has the members of `IInspectable`, then the members of this interface in declaration order. Only supported on .NET Framework. |
+
+#### COM Interface Inheritance and .NET
+
+The COM interop system that uses the <xref:System.Runtime.InteropServices.ComImportAttribute> does not interact with interface inheritance, so it can cause unexpected behavior unless the some mitigating steps are taken.
```

Comment:
```suggestion
The COM interop system that uses the <xref:System.Runtime.InteropServices.ComImportAttribute> does not interact with interface inheritance, so it can cause unexpected behavior unless some mitigating steps are taken.
```